modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APIEndpointRelevantObjectEntryModelListener.java

```diff
@@ -93,12 +93,31 @@ private void _validate(ObjectEntry objectEntry) {
 
 			String pathString = (String)values.get("path");
 
+			long apiApplicationId = (long)values.get(
+				"r_apiApplicationToAPIEndpoints_c_apiApplicationId");
+
+			if (!_objectEntryHelper.isValidObjectEntry(
+					apiApplicationId, "L_API_APPLICATION")) {
+
+				throw new ObjectEntryValuesException.InvalidObjectField(
+					null,
+					"An API endpoint must be related to an API application",
+					"an-api-endpoint-must-be-related-to-an-api-application");
+			}
+
+			long responseAPISchemaId = (long)values.get(
+				"r_responseAPISchemaToAPIEndpoints_c_apiSchemaId");
+
+			if (responseAPISchemaId != 0) {
+				_validateAPISchema(apiApplicationId, responseAPISchemaId);
+			}
+
 			if (StringUtil.equals(
 					(String)values.get("retrieveType"), "singleElement")) {
```

Comment:
Maybe we should use here [the RetrieveType constant](https://github.com/liferay/liferay-portal/blob/9e7313519afa1deaa9e1f4ca70c9a7d4d20645d7/modules/apps/headless/headless-builder/headless-builder-api/src/main/java/com/liferay/headless/builder/application/APIApplication.java#L51)

Reply:
Here we are validating a new Endpoint that is not instanced in the DB, the values of RetrieveType constants are: SINGLE_ELEMENT and COLLECTION. That's why in this case, I'm not using the constant ones.